modules/tinymce/src/plugins/table/main/ts/ui/MenuItems.ts

```diff
@@ -249,79 +272,88 @@ const addMenuItems = (editor: Editor, selections: Selections, selectionTargets:
-      editor,
-      selections,
-      Options.getTableBorderWidths(editor),
-      'tablecellborderwidth',
-      applyTableCellStyle(editor, 'border-width')
-    ),
-    onSetup: selectionTargets.onSetupCellOrRow
-  });
+    editor.ui.registry.addNestedMenuItem('tablecellborderwidth', {
+      icon: 'border-width',
+      text: 'Border width',
+      getSubmenuItems: () => buildMenuItems(
+        editor,
+        Options.getTableBorderWidths(editor),
+        'tablecellborderwidth',
+        applyTableCellStyle(editor, 'border-width')
+      ),
+      onSetup: selectionTargets.onSetupCellOrRow
+    });
 
-  editor.ui.registry.addNestedMenuItem('tablecellborderstyle', {
-    icon: 'border-style',
-    text: 'Border style',
-    getSubmenuItems: () => buildMenuItems(
-      editor,
-      selections,
-      Options.getTableBorderStyles(editor),
-      'tablecellborderstyle',
-      applyTableCellStyle(editor, 'border-style')
-    ),
-    onSetup: selectionTargets.onSetupCellOrRow
-  });
+    editor.ui.registry.addNestedMenuItem('tablecellborderstyle', {
+      icon: 'border-style',
+      text: 'Border style',
+      getSubmenuItems: () => buildMenuItems(
+        editor,
+        Options.getTableBorderStyles(editor),
+        'tablecellborderstyle',
+        applyTableCellStyle(editor, 'border-style')
+      ),
+      onSetup: selectionTargets.onSetupCellOrRow
+    });
 
-  editor.ui.registry.addToggleMenuItem('tablecaption', {
-    icon: 'table-caption',
-    text: 'Table caption',
-    onAction: cmd('mceTableToggleCaption'),
-    onSetup: selectionTargets.onSetupTableWithCaption
-  });
+    editor.ui.registry.addNestedMenuItem('tablecellbackgroundcolor', {
+      icon: 'cell-background-color',
+      text: 'Background color',
+      getSubmenuItems: () => buildColorMenu(editor, Options.getTableBackgroundColorMap(editor), 'background-color'),
+      onSetup: selectionTargets.onSetupCellOrRow
+    });
 
-  editor.ui.registry.addNestedMenuItem('tablecellbackgroundcolor', {
-    icon: 'cell-background-color',
-    text: 'Background color',
-    getSubmenuItems: () => buildColorMenu(editor, Options.getTableBackgroundColorMap(editor), 'background-color'),
-    onSetup: selectionTargets.onSetupCellOrRow
-  });
+    editor.ui.registry.addNestedMenuItem('tablecellbordercolor', {
+      icon: 'cell-border-color',
+      text: 'Border color',
+      getSubmenuItems: () => buildColorMenu(editor, Options.getTableBorderColorMap(editor), 'border-color'),
+      onSetup: selectionTargets.onSetupCellOrRow
+    });
+  }
 
-  editor.ui.registry.addNestedMenuItem('tablecellbordercolor', {
-    icon: 'cell-border-color',
-    text: 'Border color',
-    getSubmenuItems: () => buildColorMenu(editor, Options.getTableBorderColorMap(editor), 'border-color'),
-    onSetup: selectionTargets.onSetupCellOrRow
-  });
+  // TODO: TINY-8172 Unwind this when an alternative solution is found
+  if (editor.editorCommands.queryCommandSupported('mceTableToggleCaption')) {
```

Comment:
```suggestion
  if (editor.queryCommandSupported('mceTableToggleCaption')) {
```

Reply:
Alternatively... should we have a `addToggleMenuIfRegistered` function instead, as there seems to be a reasonable amount of duplication going on here?

- [ ] Consider adding a helper function for toggle menu items as well

Reply:
I added a new helper function specifically toggle menu items. Let me know if this seems alright. [`ddce24a` (#7625)](https://github.com/tinymce/tinymce/pull/7625/commits/ddce24aa0792b52a1676f504ee7329fae2d9dfc6#diff-1b110ad4c59d99bf902eac361eaaf40b035a8bda342cf38cf24746040f867133R44)

Reply:
Looks good, though I was expecting to see the same done for `Buttons.ts` as it has the same issue?

Reply:
Ah I see. Sorry I missed that. I'll the same for `addToggleButton` as well